pkg/registry/deployment/strategy.go

```diff
@@ -92,6 +92,8 @@ func (deploymentStatusStrategy) PrepareForUpdate(obj, old runtime.Object) {
 	newDeployment := obj.(*extensions.Deployment)
 	oldDeployment := old.(*extensions.Deployment)
 	newDeployment.Spec = oldDeployment.Spec
+	newDeployment.Labels = oldDeployment.Labels
```

Comment:
This strategy is for updating deployment status only (`UpdateStatus`). To update annotations/labels, `Update` should be called instead.